feat: added interceptor error

pethersonmoreno opened the pull request.

pethersonmoreno committed:
feat: added interceptor error

pethersonmoreno committed:
refactor: changed interceptor error to reject after capture the error

andrews-felipe reviewed (approved):
Pra mim a implementação está ok. Esse formato de interceptor faz com que o usuário veja as mensagens diretamente do server-side, se eleas já vierem prontas de forma amigável isso seria ótimo.

Eu só adaptaria um pouco o design do bloco pra ficar mais em conformidade com o tema do app.

o/

pethersonmoreno committed:
fix: fixed to not show snack bar error in the start of application

pethersonmoreno commented:
> Pra mim a implementação está ok. Esse formato de interceptor faz com que o usuário veja as mensagens diretamente do server-side, se eleas já vierem prontas de forma amigável isso seria ótimo.
> 
> Eu só adaptaria um pouco o design do bloco pra ficar mais em conformidade com o tema do app.
> 
> o/

@andrews-felipe , procurei outro lugar no app usando o snack bar com adaptação de visual para o tema e não encontrei nada, poderia indicar o que deve ser ajustado?

pethersonmoreno committed:
feat: changed capture error to get info about http response
 - to get error message from http response
 - and prepared it to get messages from URL and http status

pethersonmoreno committed:
feat: changed to remove auth from localstorage when refresh token is not authorized

pethersonmoreno committed:
chore: added nuxt-i18n package to project

pethersonmoreno committed:
chore: configured nuxt-i18n on project with a few translations

pethersonmoreno committed:
feat: changed message interceptor error to be translated

pethersonmoreno committed:
chore: removed duplicated nuxt config middleware: 'auth.guard'

pethersonmoreno committed:
refactor: removed not used function in forgot password
 - function name loadClientCredentials

pethersonmoreno committed:
refactor: changed to remove auth from localstorage when refresh token is not authorized

pethersonmoreno committed:
feat: changed interceptor error to show in top right like others

pethersonmoreno committed:
feat: changed message internal error to Ocorreu um erro

pethersonmoreno committed:
feat: removed custom show error messages from HTTP
 - removed because it is already show in interceptor error

pethersonmoreno committed:
Merge branch 'develop' of github.com:NewSchoolBR/newschool-frontend into feat/add-interceptor-error

filipefelice reviewed (approved):
Show, ficou muito bom! 
Único ponto que talvez mudaria seria o loading, tentaria tbm deixar a propriedade do loading genérica e setar ela pra false no interceptor mesmo que n exista, assim tirariamos todo tratamento de catch dessas telas. 
Apenas sugestão, pode ser feito em outro momento. 
👌🏻👌🏻👊🏻

pethersonmoreno commented:
@filipefelice, fiz ajustes pontuais.
Até percebi esse loading nas operações http, imagino uma outra tarefa para deixar um loading principal equivalente a exibição de errei unificada, mas mesmo assim deixaria separado
